Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiPassTree.inl

```diff
@@ -352,6 +370,26 @@ namespace AZ::Render
         }
     }
 
+    inline bool ImGuiPassTree::UpdateFilteredPassNames(AZ::RPI::Pass* pass)
+    {
+        bool anyChildMatch = m_passFilter.PassFilter(pass->GetName().GetCStr());
+
+        if (RPI::ParentPass* asParent = pass->AsParent())
+        {
+            for (const auto& child : asParent->GetChildren())
+            {
+                anyChildMatch |= UpdateFilteredPassNames(child.get());
+            }
+        }
+
+        if (anyChildMatch)
+        {
+            m_filteredPassNames.insert(pass->GetName());
```

Comment:
This set should contain the full path of the pass. The problem is with two parent passes with the same, or similar name, where a child of only one matches the filter. In this case both parent passes will be shown.